Assets/MixedRealityToolkit.Tests/PlayModeTests/ConstraintTests.cs

```diff
@@ -543,6 +543,324 @@ public IEnumerator ConstrainScaleApparentSize()
+
+            // add an xyz rotate constraint for one handed so we can only move
+            var rotateConstraint = manipHandler.EnsureComponent<RotationAxisConstraint>();
+            rotateConstraint.UseLocalSpaceForConstraint = false;
+            rotateConstraint.ConstraintOnRotation = AxisFlags.XAxis | AxisFlags.YAxis | AxisFlags.ZAxis;
+            rotateConstraint.HandType = ManipulationHandFlags.OneHanded;
+
+            // add an xyz move constraint for two handed so we can only rotate
+            var moveConstraint = manipHandler.EnsureComponent<MoveAxisConstraint>();
+            moveConstraint.UseLocalSpaceForConstraint = false;
+            moveConstraint.ConstraintOnMovement = AxisFlags.XAxis | AxisFlags.YAxis | AxisFlags.ZAxis;
+            moveConstraint.HandType = ManipulationHandFlags.TwoHanded;
+
+            yield return new WaitForFixedUpdate();
+            yield return null;
+
+            const int numHandSteps = 10;
+            
+            TestHand leftHand = new TestHand(Handedness.Left);
+            TestHand rightHand = new TestHand(Handedness.Right);
+            yield return leftHand.Show(new Vector3(-0.05f, -0.1f, 0.45f));
+            yield return rightHand.Show(new Vector3(0.05f, -0.1f, 0.45f));
+
+            // rotate and move left hand
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            yield return leftHand.SetRotation(Quaternion.Euler(45, 45, 45), numHandSteps);
+            yield return null;
+
+            TestUtilities.AssertAboutEqual(originalRotation, testObject.transform.rotation, "Rotation should be equal for one handed interaction");
+
+            yield return leftHand.Move((Vector3.left + Vector3.up) * 0.2f, numHandSteps);
+            yield return null;
+
+            TestUtilities.AssertNotAboutEqual(originalPosition, testObject.transform.position, "Position should not be equal for one handed interaction");
+
+            // return hand to original pose
+            yield return leftHand.SetRotation(Quaternion.identity, numHandSteps);
+            yield return leftHand.Move((Vector3.right + Vector3.down) * 0.2f, numHandSteps);
+            yield return null;
+
+            // grab with both hands and move/rotate
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            yield return leftHand.Move((Vector3.left + Vector3.up) * 0.2f, numHandSteps);
+            yield return null;
+
+            TestUtilities.AssertNotAboutEqual(originalRotation, testObject.transform.rotation, "Rotation should not be equal for two handed interaction");
+            TestUtilities.AssertAboutEqual(originalPosition, testObject.transform.position, "Position should be equal for two handed interaction");
+        }
+
+        /// <summary>
+        /// Tests that different constraints can apply for near and far manipulation
+        /// </summary>
+        [UnityTest]
+        public IEnumerator ConstrainByProximity()
```

Comment:
nit: i'm not a big fan of the name of this test - it's not really clear to me what it does from reading the name - CombinedConstraintFarNear?